src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -80,4 +74,60 @@ public GHClient(IProgram program)
         {
         }
     }
+
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    [ProvideService(typeof(IUIProvider), IsAsyncQueryable = true)]
+    [ProvideAutoLoad(UIContextGuids.NoSolution)]
+    [ProvideAutoLoad(UIContextGuids.SolutionExists)]
+    [Guid("D5CE1488-DEDE-426D-9E5B-BFCCFBE33E53")]
+    public sealed class ServiceProviderPackage : AsyncPackage
+    {
+        static Version vsversion;
+        public static Version VSVersion
+        {
+            get
+            {
+                if (vsversion == null)
```

Comment:
Given that `InitializeAsync` is an async method, are we sure that access to this `VSVersion` property is synchronized? I'd be worried about a race condition here, in which case I think making use of `Lazy<Version>` might be warranted.